### PR TITLE
PLAT-26773 Adding support for converting XDM fields to ParquetIOFields

### PR DESCRIPTION
**Pull request by sshefs02**

### CODE REVIEW

File: examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/field/converter/parquet/catalog/CachedCatalogSchemaParquetFieldConverter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
```
Comment (vedhera) [inline examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/field/converter/parquet/catalog/CachedCatalogSchemaParquetFieldConverter.java]: We need to add copyright header instead.

**Comment by sshefs02** [comment]
@adobe/open-source-office

File: examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/field/converter/parquet/catalog/CatalogSchemaParquetFieldConverter.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        return new ParquetIOField(
+            schemaField.getName(),
+            ParquetIODataType.GROUP,
+            ParquetIORepetitionType.OPTIONAL,
+            convert(schemaField.getSubFields())
+        );
+    }
+
+    private ParquetIOField createArrayField(SchemaField schemaField) throws ConnectorSDKException {
+        // TODO: This will blow up when support for Array of Array and Array of Map support
```
Comment (vedhera) [inline examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/field/converter/parquet/catalog/CatalogSchemaParquetFieldConverter.java]: Remove this TODO as I see support for List of List/Map will be extended seamlessly.

File: examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/writer/formatter/ParquetDataFormatter.java [hunk 3]
```diff
@@ -65,27 +65,35 @@
...
     private static Logger logger = Logger.getLogger(ParquetDataFormatter.class.getName());
 
     public ParquetDataFormatter(ParquetIOWriter writer,
                                 DataWiringParam param,
-                                ParquetFieldConverter<JSONObject> fieldConverter,
+                                ParquetFieldConverter<JSONObject> jsonFieldConverter,
                                 Extractor<JSONObject> extractor,
-                                ValidationRegistry validationRegistry) {
+                                ValidationRegistry validationRegistry,
+                                ParquetFieldConverter<List<SchemaField>> schemaFieldConverter,
```
Comment (vedhera) [inline examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/writer/formatter/ParquetDataFormatter.java]: Can we group constructor parameters for same interface together for better readability?

File: examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/writer/formatter/ParquetDataFormatter.java [hunk 4]
```diff
@@ -137,10 +145,10 @@ private boolean getRegistryEnabled() {
         try {
             // 1.Get schema from the first JSON object.
             JSONObject referenceObject = dataTable.get(0);
-            List<ParquetIOField> parquetIOFields = fieldConverter.convert(referenceObject);
+            List<ParquetIOField> parquetIOFields = jsonFieldConverter.convert(referenceObject);
```
Comment (vedhera) [inline examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/writer/formatter/ParquetDataFormatter.java]: This conversion is not required. Check original commits.

**Comment by stevengill** [comment]
Hey @sshefs02, I see that you are an Adobe employee. You need to be added as a member of this org to get that cla bot check to pass. You can add yourself by following the instructions at https://git.corp.adobe.com/OpenSourceAdvisoryBoard/handbook/blob/master/GitHub-Adobe-Org-Management.md#request-access-to-our-adobe-github-org. This way we can get your adobe ID to github ID mapping. This request will send an invitation to this org which you will have to accept. After acceptance, close and re-open the PR to trigger the check again (or just do a new commit in the pr)

**Pull request merged**